docs/Delivery-Practices/Build-and-Release/Engineering/Quality-Standards/React-Best-Practises.md

```diff
@@ -0,0 +1,45 @@
+---
+title: React Best Practises 
+authors: 
+- Rhodri Hewitson
+---
+
+## Introduction
+
+React is a popular open-source JavaScript library used for building user interfaces. It was created by Facebook and has gained a lot of popularity among developers for its declarative and efficient approach to building web applications. TypeScript is a superset of JavaScript that adds optional static typing and other features to the language. By using TypeScript with React, developers can write more reliable, maintainable, and scalable code with fewer errors. In this context, it's important to follow best practices and guidelines to ensure that your code is of the highest quality. In this post, we've provided some guidelines and best practices for writing React code in TypeScript that can help you write better code and improve the overall quality of your application.
+
+## 
+
+## Guidelines for Writing React Code in TypeScript
+
+1. Use semantic HTML: Use semantic HTML tags that describe the content of the page rather than just the appearance. This will help with accessibility and search engine optimization, and make your code easier to read and maintain.
+
+1. Use React Testing Library: Use React Testing Library to write unit and integration tests for your components. This library provides a simple and intuitive API for testing React components and encourages developers to write tests that reflect the way users interact with the application.
+
+1. Use the Atomic Design Methodology: The Atomic Design Methodology is a system for designing and building UI components in a structured and scalable way. By breaking down your UI into smaller, reusable components, you can create a more flexible and maintainable codebase.
+
+1. Use React Query: React Query is a library for managing data fetching and caching in React applications. It provides a simple and intuitive API for fetching data from APIs, and supports caching, pagination, and other advanced features.
```

Comment:
Not sure it's worth highlighting to not use `onSuccess`/`onError`/`onSettled` when using `useQuery` since it will be deprecated in future. 

https://twitter.com/TkDodo/status/1647341498227097600